### PR TITLE
IEP-465 UI menu for "idf.py python-clean" command

### PR DESCRIPTION
**Pull request by sigmaaa**
added python clean command:

Screenshot 2021-06-21 at 23 25 02

to use this command go to Project explorer > right click on project.
Test cases:

1. Run command on a built project > command successfully executed
2. Run command on a new project > command successfully executed
3. Run command on a project without build directory > user will see a message in a console that build directory is missing, nothing to clean

### CODE REVIEW

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/handlers/PythonCleanCommandHandler.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+			IStatus status = processRunner.runInBackground(arguments,
+					workDir, env);
+			if (status == null)
+			{
+				Logger.log(IDFCorePlugin.getPlugin(), IDFCorePlugin.errorStatus("Status can't be null", null)); //$NON-NLS-1$
+
+			}
+
+			// process export command output
+			exportCmdOp = status.getMessage();
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/handlers/PythonCleanCommandHandler.java:63]: Will get into NPE, if status is 'null'

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/handlers/PythonCleanCommandHandler.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		String exportCmdOp = ""; //$NON-NLS-1$
+		ProcessBuilderFactory processRunner = new ProcessBuilderFactory();
+		try
+		{
+			IStatus status = processRunner.runInBackground(arguments,
+					workDir, env);
+			if (status == null)
+			{
+				Logger.log(IDFCorePlugin.getPlugin(), IDFCorePlugin.errorStatus("Status can't be null", null)); //$NON-NLS-1$
+
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/handlers/PythonCleanCommandHandler.java:59]: Return an error message if the status is null here

**Review by kolipakakondal — CHANGES_REQUESTED** [review]
LGTM

File: bundles/com.espressif.idf.ui/OSGI-INF/l10n/bundle.properties [hunk 3]
```diff
@@ -22,4 +22,6 @@ editor.name = ESP-IDF Size Analysis
 page.name = Espressif
 
 wizard.name.1 = Espressif IDF Component
-wizard.description.0 = Create a new EspressIf IDF component
\ No newline at end of file
+wizard.description.0 = Create a new EspressIf IDF component
+command.label.2 = Python clean
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/OSGI-INF/l10n/bundle.properties:26]: Can we change it to upper case - "Python Clean"

**Comment by alirana01** [comment]
@kolipakakondal @sigmaaa I will be using this as parent for my branch to add the option to erase flash based on [IEP-462](https://jira.espressif.com:8443/browse/IEP-462)

**Comment by alirana01** [comment]
> @kolipakakondal @sigmaaa I will be using this as parent for my branch to add the option to erase flash based on [IEP-462](https://jira.espressif.com:8443/browse/IEP-462)

@kolipakakondal @sigmaaa 
Please ignore the last comment I just saw the details now

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/handlers/PythonCleanCommandHandler.java (hunk 2 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/handlers/PythonCleanCommandHandler.java]: @kolipakakondal done

File: bundles/com.espressif.idf.ui/OSGI-INF/l10n/bundle.properties (hunk 3 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/OSGI-INF/l10n/bundle.properties]: @kolipakakondal changed to ESP-IDF: Python Clean
Screenshot 2021-06-23 at 15 55 55

**Review by kolipakakondal — APPROVED** [review]
LGTM

**Pull request merged**